demux/cache.c

```diff
@@ -261,7 +261,7 @@ int64_t demux_cache_write(struct demux_cache *cache, struct demux_packet *dp)
     for (int n = 0; n < dp->avpacket->side_data_elems; n++) {
         AVPacketSideData *sd = &dp->avpacket->side_data[n];
 
-        assert(sd->size >= 0 && sd->size <= INT32_MAX);
+        assert(sd->size <= INT32_MAX);
```

Comment:
while they're pointless now they could have value if the type is ever changed in the future
the chance is tiny however so it probably doesn't make sense to keep these

Reply:
No I mean the `sd->size >= 0` check has a theoretical purpose. It stops being useless if ffmpeg changes the type in the future (very unlikely).

Reply:
I think there's two reasonable possibilities:
* don't enable this warning and drop the commit
* keep the commit as-is